### PR TITLE
Refresh linux installer scripts state

### PR DESCRIPTION
**Pull request by utzcoz**
Hi @pdsouza , I have created some patches to refresh linux installer scripts state:

1. Copy install.sh and uninstall.sh to linux idrectory from top directory. The cmd scripts have moved to windows directory, so I move it to linux directory to be consistent.
2. Fix incorrect unlocked state for sailfish(Pixel 1). In sailfish, fastboot oem device-info will show "Device unlocked" and "Device critical unlocked", so origin grep and cut command will fail to parse real device unlocked state. So I change grep content from "unlocked" to "device unlocked" to fix this problem.
3. Upgrade platform tools to 28.0.1 to support A/B updates. The 28.0.2 needs the windows updating, what maybe increase the difficulty for user, so I don't use it.

I test patches with sailfish and bullhead, and it works fine to flash images.

My work machine is Ubuntu, so I just do work for linux, not mac and windows. If needs, I can upgrade tools for windows too, based on my VirtualBox.

Thanks.

### CODE REVIEW

**Comment by pdsouza** [comment]
Thanks for the PR @utzcoz! I haven't updated these installer scripts in a while since we are now shipping the update zips directly from Maru 0.6 onwards. It is good to have the latest platform tools for Linux.

Is there any difference between `functional.sh` and `linux-functional.sh` tests? I think we can just update `functional.sh` instead since it only tests the linux install scripts?

**Comment by utzcoz** [comment]
The `linux-functional.sh` is copied from `functional.sh`, but I change the mock file path to the location in linux subdirectory. My first commit is copying `install.sh` and `uninstall.sh` to the linux subdirectory, and fix bug for incorrect locked state of `sailfish/pixel`. The `window` cmd scripts is copied to windows subdirectory, so I think maybe its better to manage `install` and `uninstall` scripts based on os type. I know macOS and Linux are both nix system, but its complicated to copy `install.sh` and `uninstall.sh` to linux or mac subdirectory to use its `adb` and `fastboot` bins to install/uninstall MaruOS manually.

**Comment by utzcoz** [comment]
@pdsouza I have changed the modification to origin `install.sh`, and squash them to a single patch. I retest it with sailfish and bullhead.

**Comment by pdsouza** [comment]
@utzcoz Thanks, I can definitely start by merging this in. 

I think you're right that we need to make it easier for users to run the scripts depending on their OS. In Maru 0.2-0.4 I had a release script that created different install packages based on the OS (one for windows, one for linux, one for mac) with the correct binaries. Maybe we can make the scripts check which platform they are running on if `adb` and `fastboot` is not in the `PATH` and use the correct binaries.

**Review by pdsouza — COMMENTED** [review]
LGTM

File: linux/mke2fs.conf [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[defaults]
```
Comment (pdsouza) [inline linux/mke2fs.conf:1]: Are these extra configuration files needed by the new platform tools?

Comment (utzcoz) [reply linux/mke2fs.conf:1]: Yes. New fastboot uses mke2fs, so I copy them to here.

Comment (utzcoz) [reply linux/mke2fs.conf:1]: We can use `grep -irn mke2fs` to check `fastboot` is using it.

Comment (pdsouza) [reply linux/mke2fs.conf:1]: OK cool, no problem. I have still been using old adb and fastboot I think so didn't know.

**Comment by pdsouza** [comment]
@utzcoz Merged, thanks! I really appreciate you taking the time to contribute to the project!

**Pull request merged**